leaderelection/context.go

```diff
@@ -174,9 +178,13 @@ func newStandardBuckets(queueName string, cc ComponentConfig) []reconciler.Bucke
 
 	bkts := make([]reconciler.Bucket, 0, cc.Buckets)
 	for name := range names {
-		bkts = append(bkts, hash.NewBucket(name, bs))
+		bkt, err := bs.NewBucket(name)
+		if err != nil {
+			return nil, err
```

Comment:
IIUC this error path is impossible to reach because the only error condition for NewBucket is when `name` is not in `names`, which we know cannot happen.

If this entire loop were a method on `BucketSet` to produce the list of buckets, then it could know this and we could avoid the unreachable error path everywhere, right?  cc @vagababov 

Unrelated to the above, it seems like the list of buckets we produce here is going to be randomly ordered, I'm not sure that's a good thing. 🤔

Reply:
You're right. Added a comment and ignore the error.

Also use a ordered list.

Reply:
IT sounds like an OK method to add. :) Iteration is the way to go.